app/controllers/api/v1/bucketlists_controller.rb

```diff
@@ -4,15 +4,17 @@ class BucketlistsController < ApplicationController
       before_action only: [:show, :update, :destroy] do
         set_bucketlist(params[:id])
       end
-      before_action :search_bucketlist, if: :query_present?, only: :index
 
       def show
         json_response(@bucketlist)
       end
 
       def index
-        @bucketlists = @current_user.bucketlists
-        @bucketlists = @bucketlists.paginate(params) if pagination_present?
+        @bucketlists = if query_present?
+          @current_user.bucketlists.search(params[:q]).paginate(params)
+        else
+          @current_user.bucketlists.paginate(params)
+        end
```

Comment:
**Code Quality - Critical**

Implement default pagination in case on no pagination params